com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/service/WatcherServiceImplTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package com.ca.lsp.cobol.service;
+
+import com.ca.lsp.cobol.service.providers.ClientProvider;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentCaptor.forClass;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+/**
+ * This class is a unit test for the {@link WatcherServiceImpl} and asserts that it creates correct
+ * JSON object structures on each watching request.
+ */
+public class WatcherServiceImplTest {
+
+  @Test
+  public void watchConfigurationChange() {
+    LanguageClient client = mock(LanguageClient.class);
+    ClientProvider provider = new ClientProvider();
+    provider.set(client);
+    ArgumentCaptor<RegistrationParams> captor = forClass(RegistrationParams.class);
+    WatcherService watcherService = new WatcherServiceImpl(provider);
+
+    watcherService.watchConfigurationChange();
+
+    verify(client).registerCapability(captor.capture());
+
+    RegistrationParams params = captor.getValue();
```

Comment:
I suggest start using, in next time,  "org.hamcrest.core" assertThat  instead of Assert methods. We can look at it in deep once we discuss in deep about TDD/BDD.  
e.g.  
assertThat(1, is(params.getRegistrations().size()));
assertThat(registration.getId(), notNullValue());

Reply:
It depends on if we have TPSR and IP checks for that

Reply:
> It depends on if we have TPSR and IP checks for that

We have it already, I even tested works fine.